src/main/java/uk/gov/ons/census/casesvc/service/CaseService.java

```diff
@@ -60,6 +70,35 @@ public Case saveCase(CreateCaseSample createCaseSample) {
     return caze;
   }
 
+  public Case buildCCSCase(
+      String caseId, SampleUnitDTO sampleUnit, String actionPlanId, String collectionExerciseId) {
+    int caseRef = getUniqueCaseRef();
+
+    Case caze = mapperFacade.map(sampleUnit, Case.class);
+    caze.setCaseRef(caseRef);
+    caze.setCaseType(sampleUnit.getAddressType());
+    caze.setCaseId(UUID.fromString(caseId));
+    caze.setActionPlanId(actionPlanId);
+    caze.setCollectionExerciseId(collectionExerciseId);
+    caze.setState(CaseState.ACTIONABLE);
+    caze.setCreatedDateTime(OffsetDateTime.now());
+    caze.setCcsCase(true);
+
+    return caze;
+  }
+
+  public Case saveCCSCaseWithUacQidLink(Case caze, UacQidLink uacQidLink) {
+    uacQidLinkRepository.saveAndFlush(uacQidLink);
+
+    caze.setUacQidLinks(new LinkedList<>(Collections.singletonList(uacQidLink)));
+    caseRepository.saveAndFlush(caze);
+
+    uacQidLink.setCaze(caze);
+    uacQidLinkRepository.saveAndFlush(uacQidLink);
```

Comment:
In fact... I think Hibernate is clever enough to save the entire case in this single `saveAndFlush`. You can lose the `caseRepository.saveAndFlush(caze)` entirely.... try it!

Reply:
@NickGrantONS, without "cascade" set on Case uacQidLinks?

Reply:
done